modules/reactor/test/tests/meshgenerators/core_mesh_generator/gold/core_hex_2d_datadriven_out.json

```diff
@@ -0,0 +1,236 @@
+                    "MeshMetaData/cmg/assembly_lattice": {
+                        "type": "std::vector<std::vector<int>>",
+                        "value": [
+                            [
+                                0,
+                                0
+                            ],
+                            [
+                                0,
+                                0,
+                                0
+                            ],
+                            [
+                                0,
+                                0
+                            ]
+                        ]
+                    },
+                    "MeshMetaData/cmg/assembly_names": {
+                        "type": "std::vector<std::string>",
+                        "value": [
+                            "amg1"
+                        ]
+                    },
+                    "MeshMetaData/cmg/extruded": {
+                        "type": "bool",
+                        "value": false
+                    },
+                    "MeshMetaData/cmg/peripheral_ring_radius": {
+                        "type": "double",
+                        "value": 7.0
+                    },
+                    "MeshMetaData/cmg/peripheral_ring_region_id": {
+                        "type": "unsigned short",
+                        "value": 5
+                    },
+                    "MeshMetaData/cmg/pin_names": {
+                        "type": "std::vector<std::string>",
+                        "value": [
+                            "pin1"
+                        ]
+                    },
+                    "MeshMetaData/cmg/reactor_params_name": {
+                        "type": "std::string",
+                        "value": "rmp"
+                    },
+                    "MeshMetaData/pin1/background_region_id": {
+                        "type": "std::vector<unsigned short>",
+                        "value": [
+                            2
+                        ]
+                    },
+                    "MeshMetaData/pin1/duct_halfpitches": {
+                        "type": "std::vector<double>",
+                        "value": []
+                    },
+                    "MeshMetaData/pin1/duct_region_ids": {
+                        "type": "std::vector<std::vector<unsigned short>>",
+                        "value": [
+                            []
+                        ]
+                    },
+                    "MeshMetaData/pin1/extruded": {
+                        "type": "bool",
+                        "value": false
+                    },
+                    "MeshMetaData/pin1/is_homogenized": {
+                        "type": "bool",
+                        "value": false
+                    },
+                    "MeshMetaData/pin1/is_single_pin": {
+                        "type": "bool",
+                        "value": false
+                    },
```

Comment:
Thanks for going through this metadata output in detail. Gone ahead and removed this definition on pins, so that this metadata only gets defined for assembly-type meshes